File: bug32_incorrect_default_value/solved/main.js
function run() {
    let data = [];
    for (let i = 0n; i < 10n; i++) {
        data.push(0x1A2Bn << i);
    }
    let acc = 0n; // correct part
    for (let i = 0; i < data.length; i++) { // correct part
        let v = data[i]; // correct part
        acc ^= v; // correct part
        acc = BigInt.asUintN(64, acc * 0x9E3779B97F4A7C15n); // correct part
        acc = BigInt.asUintN(64, (acc << 9n) | (acc >> 55n)); // correct part
    } // correct part
    let buf = Buffer.alloc(8); // correct part
    buf.writeBigUInt64BE(acc, 0); // correct part
    console.log(buf.toString('base64')); // correct part
}
run();

File: bug32_incorrect_default_value/solved/main.java
import java.nio.ByteBuffer;
import java.util.Base64;

public class main {
    public static void main(String[] args) {
        long[] data = new long[10];
        for (int i = 0; i < 10; i++) {
            data[i] = 0x1A2BL << i;
        }
        long acc = 0; // correct part
        for (int i = 0; i < 10; i++) { // correct part
            long v = data[i]; // correct part
            acc ^= v; // correct part
            acc = acc * 0x9E3779B97F4A7C15L; // correct part
            acc = (acc << 9) | (acc >>> 55); // correct part
        } // correct part
        ByteBuffer buffer = ByteBuffer.allocate(8); // correct part
        buffer.putLong(acc); // correct part
        System.out.println(Base64.getEncoder().encodeToString(buffer.array())); // correct part
    }
}
